### PR TITLE
Unify TabView layout with ambiences

### PR DESCRIPTION
**Pull request by rojkov**
Still thumbnails are squeezed when rotating TabView from landscape to portrait.

### CODE REVIEW

**Comment by rainemak** [comment]
Nice! Will give a look ASAP.

**Comment by rojkov** [comment]
Updated.
Looks like RemoveAnimation doesn't work

**Comment by rojkov** [comment]
Added target to the RemoveAnimation

File: src/pages/components/TabView.qml [hunk 1]
```diff
@@ -26,45 +26,23 @@ SilicaGridView {
...
-        bottomMargin: Theme.paddingMedium
-
-        Behavior on width {
-            enabled: !tabItem.destroying
-            NumberAnimation { easing.type: Easing.InOutQuad; duration: 150 }
-        }
-        Behavior on height {
-            enabled: !tabItem.destroying
-            NumberAnimation { easing.type: Easing.InOutQuad; duration: 150 }
+        ListView.onAdd: AddAnimation {}
```
Comment (rojkov) [inline src/pages/components/TabView.qml:45]: Is this animation visible ever? I guess it can be dropped.

**Comment by rainemak** [comment]
Found a bug from Overlay.qml that this fix revealed. 

At Overlay.qml line 414 should be like `targetPage: tabPage` not `targetPage: browserPage`

**Comment by rainemak** [comment]
I'd add now also PageHeader to the tab view. @roppola what do you think? 

Reasoning: page stack indicator gets easily hidden when the first tab has white background. There would be now space for it.

File: src/pages/components/TabItem.qml [hunk 2]
```diff
@@ -38,11 +34,8 @@ BackgroundItem {
     // thumbnail image.
     contentItem.visible: false
```
Comment (rainemak) [inline src/pages/components/TabItem.qml]: Remove this visible binding.

File: src/pages/components/TabItem.qml [hunk 3]
```diff
@@ -54,70 +47,24 @@ BackgroundItem {
 
             source: thumbnailPath
             cache: false
-            visible: false
             asynchronous: true
             smooth: true
```
Comment (rainemak) [inline src/pages/components/TabItem.qml]: Remove this as well...

File: src/pages/components/TabItem.qml [hunk 4]
```diff
@@ -54,70 +47,24 @@ BackgroundItem {
...
-            offset: 0.6
-            opacity: 1.0
-            enabled: Qt.application.active
-
-            sourceItem: roundingItem
-            anchors.fill: mask
-            direction: OpacityRamp.TopToBottom
+            gradient: Gradient {
+                GradientStop { position: 0.4; color: "transparent" }
+                GradientStop { position: 1.0; color: Theme.rgba(Theme.highlightDimmerColor, 1.0) }
```
Comment (rainemak) [inline src/pages/components/TabItem.qml]: Will check this with design.

**Comment by rojkov** [comment]
Updated

File: src/pages/components/TabItem.qml [hunk 5]
```diff
@@ -36,13 +32,9 @@ BackgroundItem {
```
Comment (rainemak) [inline src/pages/components/TabItem.qml:31]: Remove the clip couple of lines above and adjust thumbnailSize of the DeclarativeWebPage to match to the TabItem's size. There is currently unnecessary cost coming from this.

Namely:

```
        width: Screen.width - (largeScreen ? (2 * Theme.horizontalPageMargin) : 0)
        height: largeScreen ? Theme.itemSizeExtraLarge + (2 * Theme.paddingLarge) : Screen.height / 5
```

File: src/pages/components/TabItem.qml [hunk 6]
```diff
@@ -36,13 +32,9 @@ BackgroundItem {
 
     // Rounded background item that is also a placeholder for a tab not having
     // thumbnail image.
-    contentItem.visible: false
     contentItem.color: Theme.rgba(Theme.highlightBackgroundColor, Theme.highlightBackgroundOpacity)
```
Comment (rainemak) [inline src/pages/components/TabItem.qml]: Remove this one as well.

File: src/pages/components/TabItem.qml (hunk 5 above)
Comment (rainemak) [inline src/pages/components/TabItem.qml:31]: You can run browser like:

```
 QSG_VISUALIZE=overdraw sailfish-browser
```

Helps to spot extra blending. Green rectangles are good and red ones are bad.

**Comment by rojkov** [comment]
Updated with thumbnail's optimization

File: src/pages/components/WebView.qml [hunk 7]
```diff
@@ -38,9 +38,18 @@ WebContainer {
         contentItem.sendAsyncMessage(name, data)
     }
 
+    function getThumbnailSize() {
```
Comment (rainemak) [inline src/pages/components/WebView.qml]: thumbnailSize()

File: src/pages/components/WebView.qml [hunk 8]
```diff
@@ -150,7 +159,7 @@ WebContainer {
                 }
 
                 if (loaded) {
-                    webView.privateMode ? grabThumbnail() : grabToFile()
+                    webView.privateMode ? grabThumbnail(getThumbnailSize()) : grabToFile(getThumbnailSize())
```
Comment (rainemak) [inline src/pages/components/WebView.qml]: Update also FavoriteGrid.qml approx line 192. That should use always grabThumbnail(). Size could be something like Screen.width / 3 for width and height for favorites.

File: src/pages/components/WebView.qml [hunk 9]
```diff
@@ -38,9 +38,18 @@ WebContainer {
         contentItem.sendAsyncMessage(name, data)
     }
 
+    function getThumbnailSize() {
+        var largeScreen = (Screen.sizeCategory > Screen.Medium)
+        var width = Screen.width - (largeScreen ? (2 * Theme.horizontalPageMargin) : 0)
+        var height = largeScreen ? Theme.itemSizeExtraLarge + (2 * Theme.paddingLarge) : Screen.height / 5
+
+        return Qt.size(width, height)
```
Comment (rainemak) [inline src/pages/components/WebView.qml]: Design feedback: original capture could be like width of the device and height calculated with the above aspect ratio. Then in TabItem we could scale down the image to the size of the tab item.

File: src/pages/components/TabItem.qml [hunk 10]
```diff
@@ -53,71 +43,26 @@ BackgroundItem {
...
-                radius: contentItem.radius
-                color: "white"
+
+        Rectangle {
+            anchors.fill: parent
+
+            gradient: Gradient {
+                GradientStop { position: 0.4; color: "transparent" }
+                GradientStop { position: 1.0; color: Theme.rgba(Theme.highlightDimmerColor, 1.0) }
             }
```
Comment (rainemak) [inline src/pages/components/TabItem.qml:61]:
```
gradient: Gradient {
    GradientStop { position: 0.0; color: "transparent" }
    GradientStop { position: 0.45; color: Qt.rgba(0, 0, 0, 0.5)}
    GradientStop { position: 0.9; color: Qt.rgba(0, 0, 0, 0.8)}
}
```

Gradient values

File: src/pages/components/TabItem.qml [hunk 11]
```diff
@@ -53,71 +43,26 @@ BackgroundItem {
...
-            sourceItem: Rectangle {
-                x: contentItem.x
-                y: contentItem.y
-                width: contentItem.width
-                height: contentItem.height
-                radius: contentItem.radius
-                color: "white"
+
+        Rectangle {
+            anchors.fill: parent
```
Comment (rainemak) [inline src/pages/components/TabItem.qml]: With gradient we need to alter the size a bit.

```
            anchors.bottom: parent.bottom
            width: parent.width
            height: Math.max(close.height, titleLabel.height) + Theme.paddingLarge * 2
```

Note that titleLabel id doesn't exist yet, add it for title label.

File: src/pages/components/TabItem.qml [hunk 12]
```diff
@@ -144,12 +89,12 @@ BackgroundItem {
...
             anchors {
                 left: close.right
-                right: mask.right
+                right: parent.right
                 rightMargin: Theme.paddingMedium
                 verticalCenter: close.verticalCenter
             }
 
-            visible: ramp.enabled
+            font.pixelSize: Theme.fontSizeLarge
```
Comment (rainemak) [inline src/pages/components/TabItem.qml]: This needs to go back to fontSizeMedium meaning that can be removed.

File: src/pages/components/TabView.qml [hunk 13]
```diff
@@ -26,45 +26,23 @@ SilicaGridView {
...
-    readonly property int columns: portrait ? (largeScreen ? 2 : 1) : 3
-
-    cellWidth: parent.width / columns
-    // Only small screen in portrait has one column. Use then none square tab.
-    cellHeight: columns === 1 ? Screen.width / 2 : cellWidth
-
     width: parent.width
     height: parent.height
     currentIndex: -1
     header: spacer
```
Comment (rainemak) [inline src/pages/components/TabView.qml]:
```
    header: PageHeader {
        //: Tabs
        //% "Tabs"
        title: qsTrId("sailfish_browser-he-tabs")
    }
```

**Comment by rainemak** [comment]
LGTM

**Pull request merged**